### PR TITLE
Upstream configuration improvement:

### PR DESCRIPTION
**Pull request by a10zn8**
- now we can specify connector mode for better fine tuning of connection

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt [hunk 1]
```diff
@@ -119,7 +120,24 @@ open class UpstreamsConfig {
 
     class EthereumConnection : RpcConnection() {
         var ws: WsEndpoint? = null
-        var preferHttp: Boolean = false
+        var preferHttp: Boolean? = null
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt:123]: Why `preferHttp` is become nullable?

File: src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt [hunk 2]
```diff
@@ -119,7 +120,24 @@ open class UpstreamsConfig {
...
+                ConnectorMode.RPC_REQUESTS_WITH_MIXED_HEAD
+            } else {
+                if (connectorMode == null) {
+                    if (ws == null) {
+                        ConnectorMode.RPC_ONLY
+                    } else {
+                        ConnectorMode.WS_ONLY
+                    }
+                } else {
+                    ConnectorMode.valueOf(connectorMode!!.uppercase(Locale.getDefault()))
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt:137]: Maybe throw an understandable exception or just print a warning/error if there is no such value?

Comment (KirillPamPam) [reply src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt:137]: I see `parse` method in `ConnectorMode`, let's use it here

File: src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt (hunk 1 above)
Comment (a10zn8) [reply src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt:123]: we need to determine whether there was a value or not

Comment (KirillPamPam) [reply src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt:123]: I really don't see the usage of null `preferHttp`, it seems to me that non-null covers all cases

**Review by KirillPamPam — APPROVED** [review]

**Pull request merged**